components/gitpod-db/src/typeorm/migration/1704350022475-TrackEmailForFreeCredits.ts

```diff
@@ -0,0 +1,20 @@
+/**
+ * Copyright (c) 2024 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License.AGPL.txt in the project root for license information.
+ */
+
+import { MigrationInterface, QueryRunner } from "typeorm";
+import { columnExists } from "./helper/helper";
+
+export class TrackEmailForFreeCredits1704350022475 implements MigrationInterface {
+    public async up(queryRunner: QueryRunner): Promise<void> {
+        const table = "d_b_free_credits";
+        const column = "email";
+        if (!(await columnExists(queryRunner, table, column))) {
+            await queryRunner.query(`ALTER TABLE ${table} ADD COLUMN ${column} varchar(255) NULL`);
+        }
```

Comment:
I don't want to be that guy, but :see_no_evil: : What about adding an index here? This should work out of the box.
```
import { MigrationInterface, QueryRunner } from "typeorm";
import { columnExists, indexExists } from "./helper/helper";

export class TrackEmailForFreeCredits1704350022475 implements MigrationInterface {
    public async up(queryRunner: QueryRunner): Promise<void> {
        const table = "d_b_free_credits";
        const column = "email";
        const indexName = "ind_userId_email";
        if (!(await columnExists(queryRunner, table, column))) {
            await queryRunner.query(`ALTER TABLE ${table} ADD COLUMN ${column} varchar(255) NULL`);
        }

        if (!(await indexExists(queryRunner, table, indexName))) {
            await queryRunner.query(
                `ALTER TABLE \`${table}\` ADD INDEX \`${indexName}\` (userId, email), ALGORITHM=INPLACE, LOCK=NONE`,
            );
        }
    }

    public async down(queryRunner: QueryRunner): Promise<void> {}
}
```

Reply:
Whoops, it should be two indexes, as we are using them distinct. One sec...

Reply:
@svenefftinge I took the liberty to push it to this branch so you don't have to go and merge it again from some branch. Hope that's fine. :+1: 
Tests are green locally. :heavy_check_mark: